akka-stream/src/main/scala/akka/stream/Shape.scala

```diff
@@ -9,55 +9,55 @@ import scala.collection.JavaConverters._
 import scala.annotation.unchecked.uncheckedVariance
 import akka.annotation.InternalApi
 
-/**
- * An input port of a StreamLayout.Module. This type logically belongs
- * into the impl package but must live here due to how `sealed` works.
- * It is also used in the Java DSL for “untyped Inlets” as a work-around
- * for otherwise unreasonable existential types.
- */
-sealed abstract class InPort { self: Inlet[_] ⇒
-  final override def hashCode: Int = super.hashCode
-  final override def equals(that: Any): Boolean = this eq that.asInstanceOf[AnyRef]
-
-  /**
-   * INTERNAL API
-   */
-  @volatile private[stream] var id: Int = -1
-
+package impl {
```

Comment:
why not a dedicated folder? for the diff?

Reply:
Due to how `sealed` works. Classes extending `InPort`/`OutPort` need to be in the same file.

Reply:
I mean split the `Inport` and `OutPort`